### ISSUE TITLE
Subtype relations of existential metatypes are not consistent in the language.

### ISSUE THREAD

**Issue by CrystDragon** [apple/swift#70874]
### Description

For classes , if we can write code like `let a: TypeA = xxxx`, where the static type of `xxxx` is `TypeB`, it reflects the fact that `TypeB` is a subtype of `TypeA` (at least at compile time).

However, when dealing with existentials, the language seems to have inconsistent behaviors.

### Reproduction

```swift
protocol Base {}
protocol Derived: Base {}
class C: Derived {}

// case 1
let derived: Derived.Type = C.self
print(C.Type.self is Derived.Type.Type)  // true, as expected
print(C.Type.self is Derived.Type.Type)  //  compiler warning: 'is' test is always true  (1)
                                                                    // runtime: prints false                                     (2)
let derividMeta: Derived.Type.Type = C.Type.self  // compiles fine                                (3)

The behaviors of (1), (2), (3) are not consistent.

// case 2:
let base: Base.Type = derived
print(Derived.self is Base.Type) // prints false, not as expected   (4)
```

### Expected behavior

// case 1
Behaviors of (1), (2), (3) should be consistent, either they all fail or they or success. 

// case 2
I am not sure about this, but if (4) could prints true, the language surface will be more consistent in my opinion.

### Environment

* 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
* Target: arm64-apple-macosx14.0
* Xcode: Version 15.1 (15C65)

**Comment by AnthonyLatsis** [comment]
(2) looks like an issue with runtime casting.
(4) could only be true as `Derived.self is Base.Protocol` (or `(any Derived).self is (any Base).Type` as we would write it today) because `Derived.self` is a singleton metatype of type `Derived.Protocol` (`(any Derived).Type`), not an existential metatype as the one your `is` is checking against.

cc @tbkka

**Comment by CrystDragon** [comment]
> (4) could only be true as `Derived.self is Base.Protocol` (or `(any Derived).self is (any Base).Type` as we would write it today) because `Derived.self` is a singleton metatype of type `Derived.Protocol` (`(any Derived).Type`), not an existential metatype as the one your `is` is checking against.
> 
> cc @tbkka

Yes, thanks, I got it wrong. The line `let base: Base.Type = derived` can only imply that instances of `any Derived.Type` can be cast to `any Base.Type`.
